Sources/ProjectDescription/TargetMetadata.swift

```diff
@@ -0,0 +1,20 @@
+public struct TargetMetadata: Codable, Equatable, Sendable {
+    public var tags: Set<String>
+
+    public static var `default`: TargetMetadata {
+        .metadata()
+    }
+
+    /// Returns a target metadata.
+    ///
+    /// - Parameters:
+    ///   - tags: The list of tags to use. Used to select focused targets when generating the project.
+    /// - Returns: Target metadata
+    public static func metadata(
+        tags: Set<String> = []
```

Comment:
Even though a set is a more appropriate data structure, the initialization of arrays is more convenient `["foo"]` (vs `Set(["foo"])`:

```suggestion
        tags: [String] = []
```